### DISCUSSION TITLE
ECS strategies ( expert opinion request)

### DISCUSSION THREAD

**Discussion by damian-666** [bepu/bepuphysics2#247] [Ideas]
Mr. Ross sorry to bombard your vautl today but i read your SIMD vs GPGPU blog 
https://www.bepuentertainment.com/blog/2019/1/16/-but-gpus-arent-always-the-right-choice
and that wisdom is what has kept me on track and also guiding me toward my next venture  planning..  thanks for that because i was probably going to waste alot of effort on GPGPU if i didnt see that.

so Question :  im curious if  you thinkn this is an good approach for an ECS:   
https://github.com/genaray/Arch

.  im stil using an object oriented, virtual pointer, polymorphism  kind of ECS, and a  kind of old physics engine, and just planning future directives as I wrap up my current release which is performant enough for its scope  under NEt 7, but im likely  only scratchign the tip of the iceberg of the what the cpus in the field now can do for me.  even considering just a biillon cheap common android phones and pcs.   Im  new to SIMD so if you can give an opinion on this ECS im curious...

  BTW i showed to Erin Catto (box2d creator ) your blogs he liked very much your approaches.

I dont need an answer anytime soon im taking a vacation, just wondering if I should study this later, or if you like his approach.

**Comment by genaray** [comment]
> Mr. Ross sorry to bombard your vautl today but i read your SIMD vs GPGPU blog https://www.bepuentertainment.com/blog/2019/1/16/-but-gpus-arent-always-the-right-choice and that wisdom is what has kept me on track and also guiding me toward my next venture planning.. thanks for that because i was probably going to waste alot of effort on GPGPU if i didnt see that.
> 
> so Question : im curious if you thinkn this is an good approach for an ECS: https://github.com/genaray/Arch
> 
> . im stil using an object oriented, virtual pointer, polymorphism kind of ECS, and a kind of old physics engine, and just planning future directives as I wrap up my current release which is performant enough for its scope under NEt 7, but im likely only scratchign the tip of the iceberg of the what the cpus in the field now can do for me. even considering just a biillon cheap common android phones and pcs. Im new to SIMD so if you can give an opinion on this ECS im curious...
> 
> BTW i showed to Erin Catto (box2d creator ) your blogs he liked very much your approaches.
> 
> I dont need an answer anytime soon im taking a vacation, just wondering if I should study this later, or if you like his approach.

Im not Mr. Ross, neither im involved in this physics lib ( looks great btw ! )... however im one of the maintainers of https://github.com/genaray/Arch :) 

Therefore im interested in any feedback or contribution, especially if someone of you may suggest changes or new features to improve its performance even more ! If someone has questions, feel free to ask !

**Comment by RossNordby** [comment]
The high level seems reasonable at a glance. Blobs of contiguous data are indeed nice, and the benchmarks fit! (Nice work, genaray!)

I can't really do a deep dive on details, but in these sorts of situations, it's often best to just give it a try.

> BTW i showed to Erin Catto (box2d creator ) your blogs he liked very much your approaches.

I saw that! :P

**Reply by genaray** [reply]
Thanks ! Put a lot of work and love into it and there still more features coming :) Its well suited for games or data oriented programming in general and in terms of iteration speed its as fast as it can get for this architecture. Not sure about its appliance in physics however, since a hardcoded physics variant may still be faster than a general purpose ECS framework ^^
